src/mscorlib/src/System/String.cs

```diff
@@ -150,7 +186,39 @@ public static String Join<T>(String separator, IEnumerable<T> values)
             }
         }
 
+        [ComVisible(false)]
+        public static String Join<T>(Char separator, IEnumerable<T> values) {
+            if (values == null)
+                throw new ArgumentNullException("values");
+            Contract.Ensures(Contract.Result<String>() != null);
+            Contract.EndContractBlock();
+
+            using(IEnumerator<T> en = values.GetEnumerator()) {
+                if (!en.MoveNext())
+                    return String.Empty;
 
+                StringBuilder result = StringBuilderCache.Acquire();
+                if (en.Current != null) {
+                    // handle the case that the enumeration has null entries
+                    // and the case where their ToString() override is broken
+                    string value = en.Current.ToString();
```

Comment:
Yup did this one before #4382 and missed this while resolving the merge conflicts. Thanks for pointing this out. Addressed by a011339.